intermine/query.py

```diff
@@ -423,13 +424,20 @@ def from_xml(cls, xml, *args, **kwargs):
         doc = minidom.parse(f)
         f.close()
 
+        templates = doc.getElementsByTagName('template')
+        if len(templates) != 1:
+            raise QueryParseError("wrong number of templates in xml. "
+                + "Only one <template> element is allowed. Found %d" % len(templates))
```

Comment:
What query should I run to test this?

Reply:
Lines 427 and 431-432 are for extracting obj.title from the template XML.
Lines 428-430 are simple error checking, exactly analogous to that for the query XML a few lines below.